Compatibility with Eigen 3.2

Hi guys,

Cythonization of the bindings fails with Eigen 3.2 because (at least) ``AlignmentType`` is not defined:
```
Eigen3ToPython/include/eigen_wrapper.hpp:32:27: error: ‘AlignmentType’ was not declared in this scope
 using DynMap = Map<Mtype, AlignmentType::Aligned, DynStride>;
```
Best,

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,12 +9,19 @@ addons:
     packages:
     - g++
     - cmake
+    - libeigen3-dev
+env:
+  matrix:
+    - USE_SRC_EIGEN=false
+    - USE_SRC_EIGEN=true USE_EIGEN33=false
+    - USE_SRC_EIGEN=true USE_EIGEN33=true
 script:
 - . ./.travis/common.sh
 - pip install .
 - nosetests
 before_install:
-- ./.travis/dependencies/eigen-3.2
+- if [ $USE_SRC_EIGEN = true ] && [ $USE_EIGEN33 = true ]; then ./.travis/dependencies/eigen-3.3; fi
+- if [ $USE_SRC_EIGEN = true ] && [ $USE_EIGEN33 = false ]; then ./.travis/dependencies/eigen-3.2; fi
 - pip install -r requirements.txt
 notifications:
   slack:

--- a/include/eigen_wrapper.hpp
+++ b/include/eigen_wrapper.hpp
@@ -29,7 +29,13 @@ typedef Matrix<double, 6, 6> Matrix6d;
 typedef Stride<Dynamic, Dynamic> DynStride;
 
 template <typename Mtype>
+#if EIGEN_VERSION_AT_LEAST(3, 3, 0)
+using DynMap = Map<Mtype, AlignmentType::Aligned16, DynStride>;
+#elif EIGEN_VERSION_AT_LEAST(3, 2, 9)
 using DynMap = Map<Mtype, AlignmentType::Aligned, DynStride>;
+#else
+using DynMap = Map<Mtype, Aligned, DynStride>;
+#endif
 
 }
 

